util/liblog/liblog.go

```diff
@@ -61,6 +62,11 @@ func (l *lgwr) WithComponent(c string) Logr {
 	return l
 }
 
+func (l *lgwr) WithChain(c string) Logr {
+	l.l = l.l.With("chain-reference-id", c)
```

Comment:
Yes, you're correct. When I wrote this lib, I  was trying to add some sanity to the logging chaos across the project, mostly to establish fixed terms to query and build filters for. I find log correlation still massively lacking. Usually, I would often end up storing log fields in a short lived table lookup and store the hash on the context, so function calls further down the chain can easily retrieve a fully contextualised logger.

But I didn't have a lot of time when I wrote this, and just sort of built an abstraction layer to enforce some standards in logging. This has created 2 problems.

1. I cannot actually retrieve a contextualised logger from the context and have to recreate it every time i want to use it. Bad.
2.  The underlying implementation doesn't really work with a `map[string]any`, so every time I want to update a field (i.e. in a loop like you found), it appends a field instead of creating a new one.

In the future, I  want to overhaul this to allow proper contextualised logging. Until then, we need to be careful inside loops. I'll take a look at the changes.